src/cpu/arm.rs

```diff
@@ -50,61 +50,62 @@ mod abi_assumptions {
     not(target_env = "uclibc")
 ))]
 fn detect_features() -> u32 {
-    use libc::c_ulong;
-
-    // XXX: The `libc` crate doesn't provide `libc::getauxval` consistently
-    // across all Android/Linux targets, e.g. musl.
-    extern "C" {
-        fn getauxval(type_: c_ulong) -> c_ulong;
+    cfg_if::cfg_if! {
+        if #[cfg(target_arch = "aarch64")] {
+            use libc::{getauxval,AT_HWCAP, HWCAP_AES, HWCAP_PMULL, HWCAP_SHA2, HWCAP_SHA512};
```

Comment:
@BusyJay When I rebased your change on top of PR #1982, this was the only non-trivial merge conflict. Instead of using your definition of `HWCAP_SHA512`, I used the one from libc.